src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -416,6 +532,36 @@ public void Put(TStructuralType original)
             }
         }
 
+        private static IODataInstanceAnnotationContainer GetInstanceannotationContainer(PropertyInfo propertyInfo,
+        TStructuralType entity, IODataInstanceAnnotationContainer value, bool create)
+        {
+            if (entity == null)
+            {
+                throw Error.ArgumentNull("entity");
+            }
+
+            object propertyValue = propertyInfo.GetValue(entity);
+            if (propertyValue != null)
+            {
+                return (IODataInstanceAnnotationContainer)propertyValue;
+            }
+
+            if (create)
+            {
+                if (!propertyInfo.CanWrite)
+                {
+                    throw Error.InvalidOperation(SRResources.CannotSetAnnotationPropertyDictionary, propertyInfo.Name,
+                            entity.GetType().FullName);
+                }                
+
+                propertyInfo.SetValue(entity, value);
+                return value;
+            }
```

Comment:
I don't understand why there is this special logic for setting the instance annotation container within GetInstanceAnnotationContainer.  This appears to be trying to duplicate how dynamic properties work, but dynamic properties are very different -- there we are trying to set the property value within the dynamic property dictionary. Here we are trying to set the annotation container itself.